src/components/PremiumFeature/PremiumFeature.js

```diff
@@ -1,15 +1,38 @@
 import React from 'react';
 import { connect } from 'react-redux';
-import { showPremiumRequired } from '../../providers/SubscriptionProvider/SubscriptionProvider.actions';
+import {
+  updateIsInFreeCountry,
+  updateIsSubscribed,
+  updateSubscription,
+  updateIsOnTrialPeriod,
+  showPremiumRequired
+} from '../../providers/SubscriptionProvider/SubscriptionProvider.actions';
+
+function isUpdateSubscriptionNeeded(lastUpdated) {
+  if (!lastUpdated) return true;
+  const MAX_HOURS_DIFFERENCE = 12;
+  const actualTime = new Date().getTime();
+  const difference = actualTime - lastUpdated;
+  const hoursDifference = difference / (1000 * 60 * 60);
+  return hoursDifference > MAX_HOURS_DIFFERENCE;
+}
 
 function PremiumFeature({
   children,
   isOnTrialPeriod,
   isSubscribed,
   isInFreeCountry,
-  showPremiumRequired
+  showPremiumRequired,
+  lastUpdated
 }) {
   const captured = event => {
+    if (isUpdateSubscriptionNeeded(lastUpdated)) {
```

Comment:
Probably would be good rename this function by something like "isUpdateIsSubscribedNeeded" or "isUpdateSubscriberStatusNeeded"